### PR TITLE
Add error messages

### PR DESCRIPTION
**Pull request by leijurv**
Currently there are no error messages set for all the checkNotNull calls in CoinType

### CODE REVIEW

**Comment by erasmospunk** [comment]
Thanks for the contributions.

Could you make all the strings static, instead of creating them on runtime? This is for performance reasons and due to that those errors normally don't happen in production, only in development.

So for example the:

`"Coin " + name + " failed to set a unit exponent"`

could become:

`"A Coin failed to set a unit exponent"`

**Comment by leijurv** [comment]
@erasmospunk, I removed the dynamic error messages, that makes sense. But since they only will show up during development, performance doesn't matter as much.

File: core/src/main/java/com/coinomi/core/wallet/WalletPocketProtobufSerializer.java [hunk 1]
```diff
@@ -407,7 +407,7 @@ private WalletTransaction connectTransactionOutputs(Protos.Transaction txProto)
                             tx.getHashAsString(), byteStringToHash(spentByTransactionHash)));
                 }
                 final int spendingIndex = transactionOutput.getSpentByTransactionIndex();
-                TransactionInput input = checkNotNull(spendingTx.getInput(spendingIndex));
+                TransactionInput input = checkNotNull(spendingTx.getInput(spendingIndex), "Spending index " + spendingIndex + " of transaction " + spendingTx.getHashAsString() +" does not exist");
```
Comment (erasmospunk) [inline core/src/main/java/com/coinomi/core/wallet/WalletPocketProtobufSerializer.java]: Another dynamic string

File: core/src/main/java/com/coinomi/core/wallet/WalletPocket.java [hunk 2]
```diff
@@ -253,16 +253,16 @@ private void simpleAddTransaction(Pool pool, Transaction tx) {
         transactions.put(tx.getHash(), tx);
         switch (pool) {
             case UNSPENT:
-                checkState(unspent.put(tx.getHash(), tx) == null);
+                checkState(unspent.put(tx.getHash(), tx) == null, "Transaction with hash " + tx.getHashAsString() + " is already in unspent");
```
Comment (erasmospunk) [inline core/src/main/java/com/coinomi/core/wallet/WalletPocket.java]: Another dynamic string

**Comment by erasmospunk** [comment]
The strings are calculated even in normal operation, no only in development. I meant that those error almost never happen in production.
There are 2 more places to fix, otherwise looks good to merge.

**Comment by leijurv** [comment]
@erasmospunk I removed the last two error messages

**Comment by erasmospunk** [comment]
Looks good, thanks!

**Pull request merged**